### DISCUSSION TITLE
Example for Ethernet URL as stream?

### DISCUSSION THREAD

**Discussion by Zebble** [pschatzmann/arduino-audio-tools#650] [Q&A] [answered]
Are there any examples to use a URL over Ethernet as a stream instead of WiFi?

I was going to try using a TTGO T-Internet-POE board which uses a LAN8720 ethernet chip:

https://www.lilygo.cc/products/t-internet-poe

I'm trying this as I was trying to stream an MP3-based internet radio station to a Bluetooth speaker and had issues.  I'm assuming trying to use WiFi and Bluetooth is an issue as it is with other projects, so I'm hoping Ethernet will resolve.

Thanks for any advice!

**Comment by pschatzmann** [comment, marked as answer]
Just have a look at the [examples-communications](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication) directory

**Comment by Zebble** [comment]
Thanks @pschatzmann!  I did do some poking around before I posted this, and didn't find any Ethernet-specific examples.  Only WiFi or ESPNOW seemed to shown, unless I'm missing something?

**Comment by pschatzmann** [comment]
To my knowledge this uses the same TCP/IP API in Arduino...

**Comment by Zebble** [comment]
I was able to get some code going with Ethernet.  I wasn't able to get an A2DP speaker as output though, so I tried SPDIF instead.  It worked OK with WiFi (some dropouts, have not done any tuning), but kept getting ETH Connected/Disconnected events as soon as the loop started.  I'm going to try I2S instead, as soon as some MAX98357's arrive (today).

**Comment by pschatzmann** [comment]
If you don't want to wait you can test with the AnalogAudioStream and some earphones...

**Comment by Zebble** [comment]
... but I want to make things more complicated!  😄 
Yes, I will give that a try and update this discussion.
